src/main/java/hudson/plugins/gradle/BuildScanBuildWrapper.java

```diff
@@ -0,0 +1,71 @@
+package hudson.plugins.gradle;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.console.ConsoleLogFilter;
+import hudson.model.AbstractProject;
+import hudson.model.FreeStyleProject;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildWrapperDescriptor;
+import hudson.tasks.Builder;
+import jenkins.tasks.SimpleBuildWrapper;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+public class BuildScanBuildWrapper extends SimpleBuildWrapper {
+
+    @DataBoundConstructor
+    public BuildScanBuildWrapper() {
+    }
+
+    @Override
+    public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener, EnvVars initialEnvironment) throws IOException, InterruptedException {
+        // do nothing
+    }
+
+    @CheckForNull
+    @Override
+    public ConsoleLogFilter createLoggerDecorator(@Nonnull Run<?, ?> build) {
+        return new GradleConsoleLogFilter();
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) super.getDescriptor();
+    }
+
+    @Extension
+    public static final class DescriptorImpl extends BuildWrapperDescriptor {
+
+        public DescriptorImpl() {
```

Comment:
No need to add a constructor here. The class should be inferred from the outer class.